packages/react-liveness/src/components/FaceLivenessDetector/service/utils/CustomWebSocketFetchHandler.ts

```diff
@@ -0,0 +1,267 @@
+    return {
+      response: new HttpResponse({
+        statusCode: 200, // indicates connection success
+        body: outputPayload,
+      }),
+    };
+  }
+
+  /**
+   * Removes all closing/closed sockets from the socket pool for URL.
+   */
+  private removeNotUsableSockets(url: string): void {
+    this.sockets[url] = (this.sockets[url] ?? []).filter(
+      (socket) =>
+        ![WebSocket.CLOSING, WebSocket.CLOSED].includes(
+          socket.readyState as 2 | 3
+        )
+    );
+  }
+
+  private waitForReady(
+    socket: WebSocket,
+    connectionTimeout: number
+  ): Promise<void> {
+    return new Promise((resolve, reject) => {
+      const timeout = setTimeout(() => {
+        this.removeNotUsableSockets(socket.url);
+        reject({
+          $metadata: {
+            httpStatusCode: 500,
+          },
+        });
+      }, connectionTimeout);
+
+      socket.onopen = () => {
+        clearTimeout(timeout);
+        resolve();
+      };
+    });
+  }
+
+  private connect(
+    socket: WebSocket,
+    data: AsyncIterable<Uint8Array>
+  ): AsyncIterable<Uint8Array> {
+    // To notify output stream any error thrown after response
+    // is returned while data keeps streaming.
+    let streamError: Error | undefined = undefined;
+
+    // To notify onclose event that error has occurred.
+    let socketErrorOccurred = false;
+
+    // initialize as no-op.
+    let reject: (err?: unknown) => void = () => {};
+    let resolve: ({
+      done,
+      value,
+    }: {
+      done: boolean;
+      value: Uint8Array;
+    }) => void = () => {};
+
+    socket.onmessage = (event) => {
+      resolve({
+        done: false,
+        value: new Uint8Array(event.data),
+      });
+    };
+
+    socket.onerror = (error) => {
+      socketErrorOccurred = true;
+      socket.close();
+      reject(error);
+    };
+
+    socket.onclose = () => {
+      this.removeNotUsableSockets(socket.url);
+      if (socketErrorOccurred) return;
+
+      if (streamError) {
+        reject(streamError);
+      } else {
+        resolve({
+          done: true,
+          value: undefined as any, // unchecked because done=true.
+        });
+      }
+    };
+
+    const outputStream: AsyncIterable<Uint8Array> = {
+      [Symbol.asyncIterator]: () => ({
+        next: () => {
+          return new Promise((_resolve, _reject) => {
+            resolve = _resolve;
+            reject = _reject;
+          });
+        },
+      }),
+    };
+
+    const send = async (): Promise<void> => {
+      try {
+        for await (const inputChunk of data) {
+          const decodedString = this.utf8decoder.decode(inputChunk);
+          if (decodedString.includes('closeCode')) {
+            const match = decodedString.match(/"closeCode":([0-9]*)/);
+            if (match) {
+              const closeCode = match[1];
+              socket.close(parseInt(closeCode));
+            }
+            continue;
+          }
```

Comment:
This is the only new section of the file. It makes it so that if we find an event that contains a `closeCode` and a number, we send that number as the closure code for the websocket stream.

the events are received as Uint8Arrays that need to be decoded into strings